libbeat/processors/add_cloud_metadata/provider_azure_vm.go

```diff
@@ -50,11 +84,138 @@ var azureVMMetadataFetcher = provider{
 					"name": c.Str("serviceName"),
 				},
 				"region": c.Str("location"),
+				"resourceGroup": s.Object{
+					"name": c.Str("resourceGroupName"),
+				},
 			}.Apply(m)
 			return mapstr.M{"cloud": out}
 		}
 
-		fetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azSchema, azMetadataURI)
+		azGenSchema := func(m map[string]interface{}) mapstr.M {
+			orchestrator := mapstr.M{
+				"orchestrator": mapstr.M{},
+			}
+
+			orchestrator.DeepUpdate(m)
+			return orchestrator
+		}
+
+		hfetcher, err := newMetadataFetcher(config, "azure", azHeaders, metadataHost, azHttpSchema, azMetadataURI)
+		if err != nil {
+			return hfetcher, err
+		}
+		fetcher, err := newAzureMetadataFetcher("azure", hfetcher)
```

Comment:
Defined in line 44. Can you also add a comment in line 44 or here?
Basic some explanation why hfetcher, fetcher and gfetcher differ?
sth like :
hfetcher: creates the handler
fetcher: the handler with the specific cloud provider is initialised
gfethcer: the actual metadata fetcher that calls fetchAzureClusterMeta